### PR TITLE
Add PodSecurityPolicies for rancher-istio,rancher-tracing and kiali

### PR DESCRIPTION
**Pull request by bashofmann**
Add PodSecurityPolicies for rancher-istio, rancher-tracing and rancher-kiali-server.

Addresses: rancher/rancher#30744
Rebase of https://github.com/rancher/charts/pull/930

### CODE REVIEW

**Comment by haswalt** [comment]
Just hit I'm assuming this issue on my test cluster. Deployed with RKE using the permissive PSP.

Blocks any sort of ISTIO deployment on a vaguely sensible set of defaults.

Would be great to see this available ASAP

**Comment by aiyengar2** [comment]
@bashofmann I pointed your current PR back to `dev-v2.5-source` since we deprecated the `-alt` branches just now

File: packages/rancher-istio/charts/Chart.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/Chart.yaml:16]: Needs to be rebased

File: packages/rancher-istio/charts/templates/istio-install-psp.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/templates/istio-install-psp.yaml:28]: nit: newline

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
LGTM content-wise, but this needs to be rebased on the current Istio chart(s) + scripts.

@brendarearden can you help get this PR updated / merged in?

**Comment by bashofmann** [comment]
I rebased the PR and added the requested newline

File: packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml
Comment (aiyengar2) [inline packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml:65]: nit: newline

File: packages/rancher-tracing/package.yaml
Comment (aiyengar2) [inline packages/rancher-tracing/package.yaml:2]: @brendarearden can you confirm if this is the first time we’re updating rancher-tracing for this release?

**Review by aiyengar2 — DISMISSED** [review]
LGTM, nit changes suggested

File: packages/rancher-tracing/package.yaml
Comment (brendarearden) [reply packages/rancher-tracing/package.yaml:2]: correct - there have been no changes for rancher-tracing chart for 2.5.6. The changes that were needed for tracing were corrected from the istio chart. good call out!

**Review by brendarearden — DISMISSED** [review]
LGTM!! Thank you for doing this change!!

**Comment by PennyScissors** [comment]
@bashofmann @brendarearden @aiyengar2 Is this ready to merge after a rebase? I'm cleaning up this repo and just want to make sure we don't lose track of this.

**Comment by aiyengar2** [comment]
@PennyScissors let’s hold off till 2.5.6 is released.

@bashofmann sorry to ask once more, but can you rebase this PR one more time? Since this should (presumably) be targeted for 2.5.7, we’ll need to bump all the packageVersions by 1 and reset all the releaseCandidateVersions of charts that are being modified to 00.

@brendarearden could you help merge this in as soon as we’re clear to merge 2.5.7 changes so we can avoid another rebase? We should probably also check with QA / management w.r.t. whether QA has bandwidth to test this issue for the next release.

**Comment by haswalt** [comment]
Can I asked whats the reason for waiting until 2.5.7? This is currently broken for production for everybody that runs PSPs enabled. Assuming 2.5.7 will likely be a way off yet

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (aiyengar2) [inline packages/rancher-istio/charts/templates/istio-cni-psp.yaml:5]: Should this be `{{ template "istio.namespace" . }}`?

Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:5]: This happens several times across the PR, so I'll avoid spam by just adding the comment here.

File: packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml
Comment (aiyengar2) [inline packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml:4]: @brendarearden I believe that Kiali uses templating for naming resources (as I saw [here](https://github.com/rancher/charts/blob/afb287748916cdf3af3e705f3d676a7d002f5f88/charts/rancher-kiali-server/rancher-kiali-server/1.29.000-rc01/templates/deployment.yaml#L5)), so we might need to change these to use helper templates instead of literal names. 

Not sure if the same applies to Istio, but I'm not familiar enough with either chart's naming conventions to know what the right names are to use.

File: packages/rancher-tracing/package.yaml
Comment (aiyengar2) [inline packages/rancher-tracing/package.yaml:2]: @brendarearden should we be bumping the packageVersion here or should we just modify the Chart.yaml version to release a new version and reset these to 00 / 00?

@cbron this might also be a question for you: for local charts, what should our convention be w.r.t. bumping packageVersion v.s. bumping the major.minor.patch version? For pushprox, I'm leaning towards never touching the packageVersion and only touching the major.minor.patch + releaseCandidate version, re: [this comment](https://github.com/rancher/charts/pull/998#discussion_r575613219).

**Review by aiyengar2 — APPROVED** [review]
LGTM w.r.t. the content of the PR, but I have some questions regarding Istio / Tracing / Kiali conventions. Will approve and leave those choices up to Brenda.

**Comment by aiyengar2** [comment]
@haswalt we need QA bandwidth to test the contents of this PR before release (outside of a developer review) to ensure that it fully addresses all test cases in the original issue

**Comment by haswalt** [comment]
@aiyengar2 understood thank you

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (brendarearden) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:5]: Yes, this should be changed

File: packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml
Comment (brendarearden) [reply packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml:4]: yes, we should also use the kiali name templating for the resource

**Comment by brendarearden** [comment]
I am about to split istio into 1.8 and 1.9 version and I am hoping we can merge this ASAP so that both version work in an airgap environment. @bashofmann how likely is it that we could get the nits fixed and then updated with the current branch by tomorrow?

**Comment by bashofmann** [comment]
Should be doable, I have a bit of time tomorrow morning.

**Comment by bashofmann** [comment]
I replaced the hardcoded names and namespace with the appropriate templates.

File: packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml
Comment (brendarearden) [inline packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml:23]: I think this needs to match the new metadata name (this goes for all spots where you updated the kiali name stuff)

**Review by brendarearden — COMMENTED** [review]
The naming on the resources needs fixed, if you want to go ahead and fix the new line nit at the same time I think we will be good to go.

**Comment by bashofmann** [comment]
Sorry, I missed that. Done.

**Review by brendarearden — APPROVED** [review]
LGTM

**Review by aiyengar2 — APPROVED** [review]
LGTM. 

FYI @brendarearden I don't want to block this PR on nit changes, but we should eventually make changes to Istio / Kiali / Tracing to follow consistent Helm chart conventions. 

Some changes I think we should consider:
- Istio uses raw names (i.e. psp-istio-cni) across the chart, whereas tracing + kiali use helper template. I think we should move Istio towards using helper template based naming as well.
- labels like app / heritage / release should also be in a helper template, since it's copy/pasted across several resources within tracing. Maybe we should also add those labels to Istio / Kiali

Let's get this merged! 🎉

File: packages/rancher-istio/charts/Chart.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/Chart.yaml:16]: This is less for review and more for understanding: what does this annotation do?

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-cni-psp.yaml:4]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-cni-psp.yaml:26]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-cni-psp.yaml:39]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-install-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-install-psp.yaml:4]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-psp.yaml:4]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-psp.yaml:31]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-psp.yaml:46]: Wondering if we should expose this in `values.yaml`
(and its references throughout the other file(s))

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [inline packages/rancher-istio/charts/templates/istio-psp.yaml:67]: This should probably be overrideable in `values.yaml`

File: packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml
Comment (nickgerace) [inline packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml:53]: This should probably be overrideable in `values.yaml`

File: packages/rancher-tracing/charts/templates/psp.yaml
Comment (nickgerace) [inline packages/rancher-tracing/charts/templates/psp.yaml:75]: This should probably be overrideable in `values.yaml`

File: packages/rancher-istio/charts/Chart.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/Chart.yaml:16]: rancher-istio has a dependency on rancher-kiali-server, which has a CRD chart. Since this means that rancher-istio needs to install rancher-kiali-server-crd automatically on the UI in order to have access, we need to add this annotation.

This type of dependency on another package's CRD chart is not very neatly handled by the charts-build-scripts today, which is why we need to manually add this annotation.

Added an issue at https://github.com/rancher/charts-build-scripts/issues/28 to address this.

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:4]: Do you mean exposing a field that allows you to enable or disable deploying PSPs? I kinda agree with this idea since I think this is what rancher-monitoring does too:

https://github.com/rancher/charts/blob/aae93d0125fad561024b25ac1a15580ea612e158/charts/rancher-monitoring/rancher-monitoring/9.4.203/values.yaml#L330

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:26]: w.r.t. naming, I think we just need to introduce a helper template instead of hard-coding the name, re: my comment in https://github.com/rancher/charts/pull/957#pullrequestreview-615773317. Not worth blocking this PR for though.

What else would you want to expose in the values.yaml?

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-psp.yaml:67]: I'm not sure if we want to expose fields to configure the PSP used by Istio itself; is this something that other Helm charts expose today?

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (aiyengar2) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:26]: I'm hesitant to open up values.yaml fields unless absolutely necessary since it just adds more stuff we need to test / support per release.

File: packages/rancher-istio/charts/Chart.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/Chart.yaml:16]: Gotcha - thanks for the echo issue. Both the manual approach and the proposed one in the issue make sense.

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:4]: I mean both the naming and that :)
I think it may be a good idea.

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:26]: Agreed on not blocking the PR for it, but the `template` may suffice. Agreed on the `values.yaml` though; just want to anticipate potential tech debt if naming is a bottleneck or a user wants to bring their own RBAC (unlikely)

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-psp.yaml:67]: I'm moreso curious if we should expose SELinux values. This might change dependent on how the node is configured and the customer use case. However, this might be a paranoid concern

File: packages/rancher-istio/charts/templates/istio-cni-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-cni-psp.yaml:39]: Resolving this in favor of discussion: https://github.com/rancher/charts/pull/957#discussion_r597952119

File: packages/rancher-istio/charts/templates/istio-install-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-install-psp.yaml:4]: Resolving this in favor of discussion: https://github.com/rancher/charts/pull/957#discussion_r597952119

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-psp.yaml:46]: Resolving this in favor of discussion: https://github.com/rancher/charts/pull/957#discussion_r597952119

File: packages/rancher-istio/charts/templates/istio-psp.yaml
Comment (nickgerace) [reply packages/rancher-istio/charts/templates/istio-psp.yaml:31]: Resolving this in favor of discussion: https://github.com/rancher/charts/pull/957#discussion_r597952119

File: packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml
Comment (nickgerace) [reply packages/rancher-kiali-server/generated-changes/overlay/templates/psp.yaml:53]: Resolving this in favor of SELinux discussion: https://github.com/rancher/charts/pull/957#discussion_r597952976

File: packages/rancher-tracing/charts/templates/psp.yaml
Comment (nickgerace) [reply packages/rancher-tracing/charts/templates/psp.yaml:75]: Resolving this in favor of SELinux discussion: https://github.com/rancher/charts/pull/957#discussion_r597952976

**Comment by bashofmann** [comment]
Closing because PSPs have been merged with https://github.com/rancher/charts/pull/1071

**Pull request closed without merging**
